spring-cloud-starter-single-step-batch-job/src/test/java/org/springframework/cloud/task/batch/autoconfigure/JdbcItemWriterAutoConfigurationTests.java

```diff
@@ -0,0 +1,261 @@
+		applicationContextRunner.run((context) -> {
+			JobLauncher jobLauncher = context.getBean(JobLauncher.class);
+
+			Job job = context.getBean(Job.class);
+
+			JobExecution jobExecution = jobLauncher.run(job, new JobParameters());
+
+			JobExplorer jobExplorer = context.getBean(JobExplorer.class);
+
+			while (jobExplorer.getJobExecution(jobExecution.getJobId()).isRunning()) {
+				Thread.sleep(1000);
+			}
+
+			validateResultAndBean(context);
+		});
+	}
+
+	@Configuration
+	public static class TaskLauncherConfiguration {
+
+		private static Server defaultServer;
+
+		@Bean
+		public Server initH2TCPServer() {
+			Server server = null;
+			try {
+				if (defaultServer == null) {
+					server = Server.createTcpServer("-ifNotExists", "-tcp",
+							"-tcpAllowOthers", "-tcpPort", String.valueOf(randomPort))
+							.start();
+					defaultServer = server;
+					DriverManagerDataSource dataSource = new DriverManagerDataSource();
+					dataSource.setDriverClassName(DATASOURCE_DRIVER_CLASS_NAME);
+					dataSource.setUrl(DATASOURCE_URL);
+					dataSource.setUsername(DATASOURCE_USER_NAME);
+					dataSource.setPassword(DATASOURCE_USER_PASSWORD);
+					ClassPathResource setupResource = new ClassPathResource(
+							"schema-h2.sql");
+					ResourceDatabasePopulator resourceDatabasePopulator = new ResourceDatabasePopulator(
+							setupResource);
+					resourceDatabasePopulator.execute(dataSource);
+				}
+			}
+			catch (SQLException e) {
+				throw new IllegalStateException(e);
+			}
+			return defaultServer;
+		}
+
+		@Bean
+		public DataSource dataSource() {
+			DriverManagerDataSource dataSource = new DriverManagerDataSource();
+			dataSource.setDriverClassName(DATASOURCE_DRIVER_CLASS_NAME);
+			dataSource.setUrl(DATASOURCE_URL);
+			dataSource.setUsername(DATASOURCE_USER_NAME);
+			dataSource.setPassword(DATASOURCE_USER_PASSWORD);
+			return dataSource;
+		}
+
+	}
+
+	@Configuration
+	@EnableBatchProcessing
+	public static class DelimitedJobConfiguration {
+
+		@Bean
+		public ListItemReader<Map<Object, Object>> itemReader() {
+
+			List<Map<Object, Object>> items = new ArrayList<>(3);
+
+			items.add(Collections.singletonMap("item_name", "foo"));
+			items.add(Collections.singletonMap("item_name", "bar"));
+			items.add(Collections.singletonMap("item_name", "baz"));
+
+			return new ListItemReader<>(items);
+		}
+
+	}
+
+	@Configuration
+	@EnableBatchProcessing
+	public static class CustomSqlParameterSourceProviderConfiguration {
+
+		@Bean
+		public ItemSqlParameterSourceProvider<Map<Object, Object>> itemSqlParameterSourceProvider() {
+			return item -> new MapSqlParameterSource(new HashMap<String, Object>() {
+				{
+					put("item_name", item.get("item_name"));
+				}
+			});
+		}
+
+	}
+
+	@Configuration
+	@EnableBatchProcessing
+	public static class ItemPreparedStatementSetterConfiguration {
+
+		@Bean
+		public ItemPreparedStatementSetter itemPreparedStatementSetter() {
+			return new ColumnMapItemPreparedStatementSetter();
```

Comment:
How are you confirming that this is the setter used (since this is the one that gets used by default anyways)?

Reply:
If the ItemPrepareStatementSetter is set then the columnMapped method is not called.

Reply:
Updated both setters to use a Test defined setter, instead of the defaults.